File: app/src/main/java/com/example/finances/purchases/SubscribeActivity.java
package com.example.finances.purchases;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.finances.R;

public class SubscribeActivity  extends AppCompatActivity {

    private boolean isSub;

    @Override
    protected void onCreate(@Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_subscribe);

        BillingClientHelper billingClientHelper = new BillingClientHelper(this);
        isSub = billingClientHelper.isSub(BillingClientHelper.SUBSCRIPTION_MONTH);

        if(!isSub){
            Button button = findViewById(R.id.subscribe);
            button.setOnClickListener(v -> {
                billingClientHelper.subscribe(SubscribeActivity.this, BillingClientHelper.SUBSCRIPTION_MONTH);
                billingClientHelper.finish();
            });
        }
    }
}

File: app/src/main/java/com/example/finances/purchases/SubscribeActivity.kt
package com.example.finances.purchases

import BillingClientHelper
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import com.adapty.Adapty
import com.android.billingclient.api.Purchase
import com.android.billingclient.api.SkuDetails
import com.example.finances.R
import javax.inject.Inject

class SubscribeActivity : AppCompatActivity() { /*BillingClientHelper.OnPurchaseListener*/

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_subscribe)

        Adapty.getPaywalls { paywalls, products, error ->
            if (error == null) {
                var button = findViewById<Button>(R.id.subscribe_btn)
                button.text = paywalls?.get(0)?.products?.get(0)?.localizedTitle
            }
        }
    }


    /*
    @Inject
    lateinit var billingClientHelper: BillingClientHelper

    private val purchaseButtonsMap: Map<String, Button> by lazy(LazyThreadSafetyMode.NONE) {
        mapOf(
            "rsstudy.month" to findViewById(R.id.subscribe)
        )
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_subscribe)

        billingClientHelper.onPurchaseListener = this

        displayProducts()
    }

    private fun displayProducts() {
        billingClientHelper.queryProducts(object : BillingClientHelper.OnQueryProductsListener {
            override fun onSuccess(products: List<SkuDetails>) {
                products.forEach { product ->
                    purchaseButtonsMap[product.sku]?.apply {
                        text = "${product.description} for ${product.price}"
                        setOnClickListener {
                            billingClientHelper.purchase(this@SubscribeActivity, product) //will be declared below
                        }
                    }
                }
            }

            override fun onFailure(error: BillingClientHelper.Error) {
                //handle error
            }
        })
    }

    override fun onPurchaseSuccess(purchase: Purchase?) {
        TODO("Not yet implemented")
    }

    override fun onPurchaseFailure(error: BillingClientHelper.Error) {
        TODO("Not yet implemented")
    }*/
}
